cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -98,3 +104,41 @@ func runPreflightChecks(skipPreFlight bool) error {
 	fmt.Println("[preflight] Running pre-flight checks")
 	return preflight.RunRootCheckOnly()
 }
+
+// getClient gets a real or fake client depending on whether the user is dry-running or not
+func getClient(file string, dryRun bool) (clientset.Interface, error) {
```

Comment:
How is this in this commit, I've reviewed so much code, I'm positive that this is elsewhere.